src/components/pc-block-field.vue

```diff
@@ -1,432 +1,275 @@
-
-            return result
-        },
-        placeCaretAtEnd (el) {
-            if (typeof window.getSelection != "undefined"
-                    && typeof document.createRange != "undefined") {
-                var range = document.createRange();
-                range.selectNodeContents(el);
-                range.collapse(false);
-                var sel = window.getSelection();
-                sel.removeAllRanges();
-                sel.addRange(range);
-            } else if (typeof document.body.createTextRange != "undefined") {
-                var textRange = document.body.createTextRange();
-                textRange.moveToElementText(el);
-                textRange.collapse(false);
-                textRange.select();
-            }
-        }
+    fieldWrapperClasses() {
+      const obj = {}
+
+      obj['pc-field--read-only'] = this.isReadOnly
+      obj['pc-field--focused'] = this.isFocused
+      obj['pc-field-' + this.fieldProp] = true
 
+      return obj
     },
-    watch: {
-        val (newValue, oldValue) {
-            let newVal = parseFloat(this.formatNumberFields(newValue)),
-                oldVal = parseFloat(this.formatNumberFields(oldValue));
-
-            if (this.isReadOnly || !this.isFocused || (newVal == oldVal)) {
-                return;
-            }
-
-            // updating calculations
-            this.$store.dispatch('field:change', {
-                prop: this.fieldProp,
-                value: newVal || 0
-            })
-        },
-        fieldValue () {
-            // in case some input field on the same block changes the main math values
-            // we need to update the input fiel
-            let anotherFieldInBlockIsUpdating = !this.isFocused;
-            if (!this.isReadOnly && !anotherFieldInBlockIsUpdating) {
-                return
-            }
-
-            this.val = this.fieldValue;
-            if (this.enableEdit) {
-                this.$refs['pc-value'].textContent = this.val;
-            }
-        },
-        isFocused (newValue) {
-            this.$emit('update:focus', {
-                fieldProp: this.fieldFromBlock || this.fieldProp,
-                value: newValue
-            })
-        }
+    fieldFormattedStyle() {
+      const result = {}
+      if (this.isFocused) {
+        result.display = 'none'
+      }
+
+      return result
+    },
+    fieldEditableStyle() {
+      const result = {}
+      if (!this.isFocused) {
+        result.opacity = 0
+      }
+
+      return result
     },
-    directives: {
-        initialvalue: {
-            inserted (el, directive, vnode) {
-                el.textContent = vnode.context.val
-            }
+    testType() {
+      return this.$store.getters.testType
+    },
+  },
+  methods: {
+    getSanitizedPcValue() {
+      // People will use copy paste. We need some data sanitization
+
+      // remove markup
+      const oldValue = this.$refs['pc-value'].textContent + ''
```

Comment:
Why are you adding an empty string? Isn't the textContent inherently a string?

Reply:
It is a nullable field, so the purpose is to ensure oldValue is always a string. However, I just checked and `null + ''` equals `'null'` -.-